### PR TITLE
feat: discord-module to database

### LINKED ISSUE #96: Connect discord-module to database

### PR DESCRIPTION
**Pull request by Cahllagerfeld**
closes #96 

- [x] actual functionality
- [x] unit tests
- [x] e2e tests

### CODE REVIEW

File: src/discord/discord.service.ts [hunk 1]
```diff
@@ -1,44 +1,74 @@
...
 
 @Injectable()
 export class DiscordService {
-  private discords: DiscordProfile[] = [];
+  constructor(private readonly astraService: AstraService) {}
 
-  create(createDiscordDto: DiscordDTO): DiscordProfile {
+  create(createDiscordDto: DiscordDTO): Observable<documentId> {
     const discordUser = {
       id: 123,
```
Comment (eddiejaoude) [inline src/discord/discord.service.ts:19]: Is the temporary `id` still needed?

File: test/features/discord.feature [hunk 2]
```diff
@@ -1,101 +1,112 @@
...
-#     Then the response status code should be 200
-#     And the response should contain:
-#       | id        | 123                        |
-#       | bio       | "created user bio"         |
-#       | username  | "create-user"              |
-#       | socials   | {"discord":"create-user"}  |
-#       | updatedOn | "2021-01-01T00:00:00.000Z" |
-#       | createdOn | "2021-01-01T00:00:00.000Z" |
 
+    Scenario: get users with authenticated request
```
Comment (eddiejaoude) [inline test/features/discord.feature:86]: I think this should be singular

```suggestion
    Scenario: get user with authenticated request
```

File: test/features/discord.feature [hunk 3]
```diff
@@ -1,101 +1,112 @@
...
-#       | username | "delete-user"             |
-#       | socials  | {"discord":"delete-user"} |
-#     When make a DELETE request to "/discord/321"
-#     Then the response status code should be 404
-#     And the response should contain:
-#       | statusCode | 404              |
-#       | message    | "User Not Found" |
+    Scenario: delete non-existing user
+        Given authorisation
+        And make a POST request to "/discord" with:
```
Comment (eddiejaoude) [inline test/features/discord.feature:72]: I don't think we need to create a user for this test

File: test/features/discord.feature [hunk 4]
```diff
@@ -1,101 +1,112 @@
...
-#       | createdOn | "2021-01-01T00:00:00.000Z" |
+    Scenario: update a user
+        Given authorisation
+        And make a POST request to "/discord" with:
+            | bio      | "This is a GitHub Campus Expert"                                                              |
+            | username | "khattakdev"                                                                                  |
+            | socials  | {"discord":"khattakdev","github":"khattakdev","linkedin":"khattakdev","twitter":"khattakdev"} |
+        Then the response should contain:
+            | documentId | "TYPE:ID" |
+        When make a PUT request to "/discord/{id}" with:
```
Comment (eddiejaoude) [inline test/features/discord.feature:51]: what is the value of `{id}`?

File: test/features/discord.feature [hunk 5]
```diff
@@ -1,101 +1,112 @@
...
+            | bio      | "Updated user bio"        |
+            | username | "updated-user"            |
+            | socials  | {"discord":"update-user"} |
+        Then the response status code should be 200
+        And the response should contain:
+            | documentId | "TYPE:ID" |
```
Comment (eddiejaoude) [inline test/features/discord.feature:57]: the updated fields should be re-checked

**Comment by eddiejaoude** [comment]
Looks good 👍 

I left a few comments, not sure if these are similar in other modules?

File: test/features/discord.feature (hunk 4 above)
Comment (Cahllagerfeld) [reply test/features/discord.feature:51]: `id` is the Id of the document that got created in the step before. This needs to be dynamically

Comment (eddiejaoude) [reply test/features/discord.feature:51]: Ah yes, the step def checks for this 👍

**Review by eddiejaoude — APPROVED** [review]
Looks great 👍

**Pull request merged**